test/directory_watcher/mac_os_test.dart

```diff
@@ -25,36 +23,31 @@ void main() {
 
   test(
       'does not notify about the watched directory being deleted and '
-      'recreated immediately before watching', () {
+      'recreated immediately before watching', () async {
     createDir("dir");
     writeFile("dir/old.txt");
     deleteDir("dir");
     createDir("dir");
 
-    startWatcher(path: "dir");
+    await startWatcher(path: "dir");
     writeFile("dir/newer.txt");
-    expectAddEvent("dir/newer.txt");
+    await expectAddEvent("dir/newer.txt");
   });
 
   test('emits events for many nested files moved out then immediately back in',
-      () {
+      () async {
     withPermutations(
         (i, j, k) => writeFile("dir/sub/sub-$i/sub-$j/file-$k.txt"));
 
-    startWatcher(path: "dir");
+    await startWatcher(path: "dir");
 
     renameDir("dir/sub", "sub");
     renameDir("sub", "dir/sub");
 
-    allowEither(() {
-      inAnyOrder(withPermutations(
-          (i, j, k) => isRemoveEvent("dir/sub/sub-$i/sub-$j/file-$k.txt")));
+    await inAnyOrder(withPermutations(
+        (i, j, k) => isRemoveEvent("dir/sub/sub-$i/sub-$j/file-$k.txt")));
 
-      inAnyOrder(withPermutations(
-          (i, j, k) => isAddEvent("dir/sub/sub-$i/sub-$j/file-$k.txt")));
-    }, () {
-      inAnyOrder(withPermutations(
-          (i, j, k) => isModifyEvent("dir/sub/sub-$i/sub-$j/file-$k.txt")));
-    });
+    await inAnyOrder(withPermutations(
+        (i, j, k) => isAddEvent("dir/sub/sub-$i/sub-$j/file-$k.txt")));
```

Comment:
Why isn't this using `allowEither()` anymore?

Reply:
Just because it does so in your local tests doesn't mean it *always* does so. These alternate cases came about through painstaking debugging of intermittent edge cases that crop up in specific contexts. It's not safe to just delete them because the tests pass locally.

Reply:
Added back the backup cases.

Reply:
It looks like this backup case is still missing.

Reply:
Fixed.